### PR TITLE
ocamlPackages.ocamlformat-mlx: init at 0.26.2.0

### PR DESCRIPTION
**Pull request by Denommus**
ocamlformat-mlx is a fork of ocamlformat to format .mlx files (OCaml files with jsx-like syntax). I'm doing it in a different directory from ocamlformat altogether because it has both a different versioning system and a different src from ocamlformat, so I didn't see a reason to try to adapt it to the `generic.nix` file in ocamlformat.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

- **Extra args**: ``
- **Free space**: true
- **Who asked**: lucasew

**TARGETS**
- **x86_64-linux**: true
- **x86_64-darwin**: true
- **aarch64-darwin**: true

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12449825647

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
  </ul>
</details>

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
  </ul>
</details>

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
  </ul>
</details>

**Review by anmonteiro — APPROVED** [review]
works for me. I would just rename the files `lib.nix` and `default.nix` to match the convention in the other packages.

**Comment by Denommus** [comment]
@anmonteiro I'm following the naming convention of ocamlformat

**Comment by Denommus** [comment]
@anmonteiro I did what you suggested

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12496230944


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12496617554

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12496617554


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>hugo</li>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>ocamlPackages.ocamlformat-mlx</li>
    <li>ocamlPackages.ocamlformat-mlx-lib</li>
  </ul>
</details>

**Comment by lucasew** [comment]
I guess ill have to add `--eval local` in default workflow args
Screenshot_20241225-203406

File: pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  buildDunePackage,
+  fetchFromGitHub,
+  cmdliner,
+  odoc,
+  ocamlformat-mlx-lib,
+  re,
+}:
+buildDunePackage rec {
+  pname = "ocamlformat-mlx";
+  version = "0.26.2.0";
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix:12]: We likely want inherit things from ocamlformat-mlx-lib to make sure they are not out of sync.
`inherit (ocamlformat-mlx-lib) version src`

File: pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  buildInputs = [
+    cmdliner
+    re
+    odoc
+    ocamlformat-mlx-lib
+  ];
+
+  meta = {
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix:29]: Meta also seems to be the same as `ocamlformat-mlx-lib`

File: pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix (hunk 1 above)
Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/ocamlformat-mlx/default.nix:12]: Not sure if minimalOcamlVersion is inheritable as well

**Comment by Denommus** [comment]
@ulrikstrid done

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367235`

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**